### DISCUSSION TITLE
Need help calling color picker using onclick on a span.

### DISCUSSION THREAD

**Discussion by itpaksolution** [mdbassit/Coloris#124] [Q&A] [answered]
So my task is to make color picker work on Mac OS and Safari iPhone.
The thing is: with this color picker I can't find doc on how to show it from jquery using onclick="..." in my span.
And no, I can't use input field.

Also I should mention that the default input field ain't working. Pls help, I love your solution

**Comment by mdbassit** [comment]
Coloris does not support attaching the color picker to a span. It has to be an input field, which you can style to make it look like anything you want with CSS.

> Also I should mention that the default input field ain't working. Pls help, I love your solution

Can you explain what doesn't work exactly? Can you share the code you use?

**Reply by itpaksolution** [reply, marked as answer]
Can't share the code because of NDA. But I fixed default color picker with 4 lines of code.
The thing was that the color type input field on iphone and mac can't be 'display:none;' or 0 sizes. So the solution was to make it work, and that solution is:
`
    display: block !important;
    opacity: 0 !important;
    visibility: hidden !important;
    position: absolute !important;
`
